prime-router/docs/authz/authz-transition-plan.md

```diff
@@ -5,35 +5,78 @@ This document will detail the steps needed to successfully transition from the c
+### Integrate auth and submissions microservices to CI processes ###
+We should begin including the microservices projects in our continuous integration builds, set up the API endpoints in
+the staging environment, and make sure the microservices are executed. This will also require setting up environment-
+specific secrets for connecting with Okta and ensuring we have set up permanent application users for the microservices.
+
+### Update end to end testing and development environments to include auth and submissions ###
+We should build a new test based on `end2end_up` that performs submission through the microservices instead of the
+functionapp. We will need to determine if this test should integrate with Okta or if Okta connections should be mocked
+in the absence of an offline test container.
+
+### Create application users for senders and begin migration outreach ###
+Senders are represented as application users. Creating the application user will produce a client ID and a private key
+that the sender will use to authenticate to the microservices. Begin outreach to at least one sender to have them begin
+using the microservices to submit instead of the functionapp.
+
 ## Migration of senders and website users ##
 * Migrate all senders in staging
-  * Can existing public keys be transferred to Okta, or will all senders have to be fully reonboarded?
+  * Can existing public/private keys be transferred to Okta, or will all senders have to be fully reonboarded?
   * Do we intend to run both auth systems simultaneously?
+    * Document differences between RS auth and proposed Okta auth (endpoints used, etc.) 
 * Users: Process to migrate existing user authorizations
     * Map existing RS scopes to Okta scopes
     * How to restrict users to information related to their organization
 * Do sender and user migrations need to be performed simultaneously?
 
 ## Other Questions ##
 
-Do we need to maintain both auth systems simultaneously?
+Should we maintain old and new auth/report endpoints simultaneously?
 
-* Pros:
-  * Flexible timeline for transitioning senders; don't have to migrate all at once
-  * Can designate a pilot partner
+* Pros for maintaining separate endpoints:
+  * Flexible timeline for transitioning senders; no hard cutoff date, don't have to migrate all at once
+    * Coordinating a cutover date with external partners may not be feasible
+  * Can designate a pilot partner before initiating a larger migration
+    * Good idea to have a pilot partner even if not preserving separate endpoints for new API 
 
 * Cons:
   * Transparent cutover not an option; new API must have different path
-  * Will require tests for both APIs to be maintained simultaneously
+  * Will require implementation and tests for both APIs to be maintained simultaneously
+  * Will likely require some level of reonboarding for all users of RS
   * Can both APIs coexist on the same listening port? Would this require the functionapp to act as a passthrough?
 
 
```

Comment:
Also wanted to leave some musings on the SMART on FHIR question below. 

Context: Two jobs ago I worked on an EHR vendor's FHIR api and a sibling team of ours handled our companies SMART on FHIR implementation.

Overall, from what I recall, SMART on FHIR is irrelevant to our use case because it's mainly for a seamless auth workflow between apps. Ex: If a clinic was using Cerner or Epic's EHR system and wanted to send a report to report stream directly to us without an intermediary. So they would use the EHR to launch a report stream SMART on FHIR app and the authorization handshake would happen at that point. 

There's probably other use cases for it but just wanted to throw out my 2 cents.